src/rust/engine/src/cffi_build.rs

```diff
@@ -18,10 +18,37 @@ native engine binary, allowing us to address it both as an importable python mod
 
 */
 
+use std::env;
+use std::fs;
+use std::io::{Read, Result};
+use std::path::PathBuf;
+
 fn main() {
-  gcc::Config::new()
-    // N.B. The filename of this source code - at generation time - must line up 1:1 with the
-    // python import name, as python keys the initialization function name off of the import name.
-    .file("src/cffi/native_engine.c")
-    .compile("libnative_engine_ffi.a");
+  let mut config = gcc::Config::new();
+
+  let path = PathBuf::from("src/cffi/native_engine.c");
+
+  // N.B. The filename of this source code - at generation time - must line up 1:1 with the
+  // python import name, as python keys the initialization function name off of the import name.
+  config.file(path.to_str().unwrap());
+  for flag in make_flags(&path).unwrap() {
+    config.flag(flag.as_str());
+  }
+
+  config.compile("libnative_engine_ffi.a");
+}
+
+fn make_flags(c_file: &PathBuf) -> Result<Vec<String>> {
```

Comment:
Because of the owned/borrowed split, `&PathBuf` should almost always be `&Path` instead, as it is less restrictive for the caller (likewise: almost always `&str` rather than `&String`).

Reply:
Done